pino.js

```diff
@@ -35,10 +37,27 @@ function pino (opts, stream) {
   var level = opts.level || 'info'
   var serializers = opts.serializers || {}
   var end = ',"v":' + LOG_VERSION + '}\n'
-  return new Pino(level, stream, serializers, stringify, end, name, hostname, slowtime, '')
+  var cache = opts.extreme ? 4096 : 0
+
+  var logger = new Pino(level, stream, serializers, stringify, end, name, hostname, slowtime, '', cache)
+
+  if (cache) {
+    onExit(function () {
+      if (stream === process.stdout) {
+        console.log(logger.buf)
+      } else if (stream === process.stderr) {
+        console.error(logger.buf)
+      } else {
+        stream.write(logger.buf)
+      }
+      process.exit()
```

Comment:
are you sure this is ok? We are losing the exit code, and possibly interacting with other handlers. Maybe we should do it the other way, let people to add some code to their app? I'm especially concerned about all the SIG* handlers, because if I have some logic added there to do some things when these happens, the process will end up exiting.